### ISSUE TITLE
Web Development with Python

### ISSUE THREAD

**Issue by Rick476** [golang/vscode-go#4024]
import pandas as pd
from sklearn.preprocessing import LabelEncoder, StandardScaler
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split

# Step 1.3: Load the dataset
customer_data = pd.read_csv('customer_data.csv')

# Step 1.4: Fill missing tenure values with the mean
customer_data['tenure'] = customer_data['tenure'].fillna(customer_data['tenure'].mean())

# Step 1.5: Encode the 'state' column
state_encoder = LabelEncoder()
customer_data['encoded_state'] = state_encoder.fit_transform(customer_data['state'])

# Step 1.6: Scale tenure and monthly_charges
scaler = StandardScaler()
customer_data[['tenure', 'monthly_charges']] = scaler.fit_transform(
    customer_data[['tenure', 'monthly_charges']]
)

# Train/test split
X = customer_data[['tenure', 'monthly_charges', 'encoded_state']]
y = customer_data['churn']

x_train, x_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

# Train the model
model = LogisticRegression()
model.fit(x_train, y_train)

# Predictions
churn_probabilities = model.predict_proba(x_test)[:, 1]

# Dashboard calculations
avg_churn_prob = churn_probabilities.mean()
high_risk_customers = (churn_probabilities > avg_churn_prob).sum()

# Attach states to test set predictions
test_states = customer_data.loc[x_test.index, 'state']
state_probs = pd.DataFrame({
    'state': test_states.values,
    'prob': churn_probabilities
})

# Compute churn rate + high risk count per state
state_results = []
for state in sorted(customer_data['state'].unique()):
    subset = state_probs[state_probs['state'] == state]['prob']
    churn_rate = subset.mean()
    high_risk_count = (subset > avg_churn_prob).sum()
    state_results.append((state, churn_rate, high_risk_count))

# Write EXACTLY 17 lines
with open('churn_results.txt', 'w') as f:
    f.write(f"{avg_churn_prob}\n")
    f.write(f"{high_risk_customers}\n")
    for state, churn_rate, high_risk_count in state_results:
        f.write(f"{state},{churn_rate},{high_risk_count}\n")
